block_scheduler.go

```diff
@@ -0,0 +1,139 @@
+package simplex
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+var ErrTooManyPendingVerifications = errors.New("too many blocks being verified to ingest another one")
+
+type BlockVerificationScheduler struct {
```

Comment:
Would it make sense to just call it BlockDependencyManager or BlockDependencies or something similar? A struct should be named according to its operations, not according to how it works.